components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -1816,7 +1816,7 @@ public ServiceProvider getApplication(String applicationName,
             localServiceProvider.setDescription("Local Service Provider");
             applicationId = createServiceProvider(tenantDomain, localServiceProvider);
         }
-        return getApplication(applicationId);
+        return getApplication(applicationId, tenantDomain);
```

Comment:
Isn't the application id unique across the system?

Reply:
Yes. It is unique. The problem was the `getApplication` method use a tenant domain (taken from context) to retrieve service provider details. Ideally, it seems the sub sequent methods which use application ID and tenant domain should be modified to only have the application Id.
https://github.com/wso2/carbon-identity-framework/blob/master/components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java#L2147~L2165